polaris.shopify.com/src/components/TOC/TOC.tsx

```diff
@@ -0,0 +1,145 @@
+import { useEffect, useRef, useState } from "react";
+import { TOCItem } from "../../utils/hooks";
+import { className } from "../../utils/various";
+import styles from "./TOC.module.scss";
+
+interface Props {
+  items: TOCItem[];
+}
+
+function getContentTopMargin(): number {
+  const rootStyles = getComputedStyle(document.documentElement);
+  const headerHeight = rootStyles.getPropertyValue("--header-height");
+  const headerMargin = rootStyles.getPropertyValue("--header-margin");
+  const headerThreshold = parseInt(headerHeight) + parseInt(headerMargin);
+  return headerThreshold;
+}
+
+function scanPageForCurrentHeading(): string | void {
+  const contentTopMargin = getContentTopMargin();
+  const headings = document.querySelectorAll("#main h2, #main h3");
+  let currentHeading: Element | null = null;
+
+  // Scan the headings from the bottom. The heading that comes first
+  // after the "content top margin" is the current heading.
+  for (let i = headings.length - 1; i >= 0; i--) {
+    const heading = headings[i];
+    const { top } = heading.getBoundingClientRect();
+    currentHeading = heading;
+    if (top < contentTopMargin) {
+      break;
+    }
+  }
```

Comment:
The thing is, the browser doesn't take the fixed heading into account, so it awkwardly scrolls so that the target heading is hidden below the header.

Reply:
Ok. I don't know why the heading is fixed?

Reply:
1. Highly subjective, but I quite like the UX of it. Makes the site feel a bit more app-like.

2. To make the side navs scrollable, they need to have an overflow and a fixed height. Having a fixed header makes the height of the sidebars consistent and predictable (`calc(100vh - var(--header-height))`). Without a fixed header, their heights will range from `calc(100vh - var(--header-height))` (header is visible) and `100vh` (header is not visible). It adds another type of headache.

You can see how the not-fixed Vercel bar here messes with the sidebars and makes them cut off in the middle of the air. I want to avoid that.

https://user-images.githubusercontent.com/875708/179785550-e3bb9f06-9ffc-4a45-a1f3-add96a7dd06b.mov